src/cmarkit.ml

```diff
@@ -42,9 +42,9 @@ module Block_line = struct
     in
     loop s [] (String.length s - 1) 0 0
 
-  let flush ?(meta = Meta.none) s start last acc = match acc with
-  | [] -> [s, meta]
-  | acc -> (String.sub s start (last - start + 1), meta) :: acc
+  let flush ?(meta = Meta.none) s start last acc =
+    let sub = String.sub s start (last - start + 1) in
+    (sub, meta) :: acc
```

Comment:
So in fact this function is not directly or indirectly used by code of the code base right ? 

I think the intent was to write (see `flush_tight`): 

```ocaml
if start > last then (s, meta) :: acc else
let sub = String.sub s start (last - start + 1) in
(sub, meta) :: acc
```
to avoid the allocation (AFAIR think `String.sub` never integrated that strings became immutable since you may still configure the compiler so that they are not). But the mind was somewhere else apparently.

Could you please switch the code to that.

Reply:
What I wrote above is non-sense. You need to check start = 0 and last = String.length s. Used what you proposed.